README.md

```diff
@@ -1,52 +1,49 @@
 
 This project gets rid of the GDB layer and provides a direct interface to the debug probe,
-which then enables other software, for example [VisualStudio](https://code.visualstudio.com/blogs/2018/08/07/debug-adapter-protocol-website) to use it's debug functionality.
+which then enables other software to use it's debug functionality.
 
-What's more is that we can use CoreSight to its full extent. We can trace and modify memory as well as registers in real time.
-
-**The end goal is a complete library toolset to enable other tools to use the functionality of CoreSight.**
+**The end goal of this project is to have a complete library toolset to enable other tools to communicate with embedded targets.**
 
 ## Functionality
 
-- The lib can connect to a DAPLink or an STLink and read and write memory.
-- It can read ROM tables and thus extract CoreSight component information.
-- It can download ELF binaries using standard ARM flash blobs.
-- Basic debugging (attach, reset, halt, step, show stacktrace, add breakpoint, halt on breakpoint) works.
+As of 0.6.0 this lib can
 
-Focus of the development is having a full implementation (CoreSight, Flashing, Debugging) working for the DAPLink and go from there.
+- connect to a DAPLink, STLink or JLink
+- talk to ARM and Risc-V cores via SWD or JTAG
+- read and write arbitrary memory of the target
+- halt, run, step, breakpoint and much more the core
+- download ELF, BIN and IHEX binaries using standard CMSIS-Pack flash algorithms to ARM cores
+- provide debug information about the target state (stacktrace, stackframe, etc)
 
-### Downloading a file
+To see what new functionality was added have a look at the [CHANGELOG](CHANGELOG.md)
 
-The `cargo-flash` utility can be used as a cargo subcommand to download a compiled Rust program onto a target device. See https://github.com/probe-rs/cargo-flash for more information.
+## Downloading a file
 
+The `cargo-flash` utility can be used as a cargo subcommand to download a compiled Rust program onto a target device. It can also be used to download arbitrary ELF files that might come out of a C/C++ compiler. Have a look at [cargo-flash](https://github.com/probe-rs/cargo-flash) for more information.
 
-### CLI
+## GDB
 
-To demonstrate the functionality a small cli was written.
-Fire it up with
+We provide a GDB stub you can use until [Microsoft DAP](https://code.visualstudio.com/blogs/2018/08/07/debug-adapter-protocol-website) support is fully implemented.
+You can find it [here](https://github.com/probe-rs/probe-rs/tree/master/gdb-server) and you can also use it from within `cargo-flash` with the `--gdb` flag.
 
-```
-cargo run -p probe-rs-cli -- help
-```
+## VScode
 
-The help dialog should then tell you how to use the CLI.
+We are implementing [Microsoft DAP](https://code.visualstudio.com/blogs/2018/08/07/debug-adapter-protocol-website) to provide full probe-rs integration into modern debuggers such as the built in one of VSCode.
```

Comment:
If we mention this here, I wouldn't mention it in the GDB section. Also, the debugger is not really in VS Code, the main part will be in probe-rs. I would also write VS Code the same everywhere, in the section heading and here.

Reply:
This is not in the GDB section, there is a VSCode section inbetween ;)

Reply:
This is the VS Code section, yes, but Microsoft DAP is also mentioned in the GDB section.

Reply:
Ah, whoops, sorry, didn't understand that properly ...